worlds/alttp/ItemPool.py

```diff
@@ -651,6 +652,73 @@ def place_item(loc, item):
             place_item(key_location, item_to_place)
         else:
             pool.extend([item_to_place])
+
+    dungitemreplacements = diff.extras[0] + diff.extras[1] + diff.extras[2] + diff.extras[3] + diff.extras[4]
+    world.random.shuffle(dungitemreplacements)
+
+    if world.smallkey_shuffle[player] == smallkey_shuffle.option_start_with:
```

Comment:
Instead of a large explicit list duplicating counts (and therefore making it easy to accidentally not update both spots), would it not make more sense to iterate over the Dungeon items, filter by .type == start with type and grant those items? As in move most of this to make_dungeon.